### PR TITLE
test: Change default catalog to persist

### PR DESCRIPTION
**Pull request by jkosh44**
There are some tests that cannot use the shadow catalog for various
reasons. Previously, those tests defaulted to using the stash catalog,
because that is the catalog used in production. We are actively working
on switching the stash catalog to the persist catalog in production,
and we're not changing the stash catalog at all. Therefore, we'd prefer
to test the persist catalog over the stash catalog.

Works towards resolving #24218

### Motivation
Update tests

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Comment by jkosh44** [comment]
Nightlies: https://buildkite.com/materialize/nightlies/builds/6409
Release qualification: https://buildkite.com/materialize/release-qualification/builds/426

**Comment by jkosh44** [comment]
@MaterializeInc/testing I'm not sure which nightly failures are expected and which are new.

**Comment by nrainer-materialize** [comment]
> @MaterializeInc/testing I'm not sure which nightly failures are expected and which are new.

Thanks for reaching out and triggering a nightly build.
Two jobs failed in the referenced build; both due to `thread 'coordinator' panicked at src/catalog/src/durable/impls/stash.rs:1201:13: global timestamp must always go up`. This is resp. was a known issue and tracked with https://github.com/MaterializeInc/materialize/issues/25075 (fixed on latest main now). That is, the failure is unrelated to your changes; you are good to proceed with the merge!

**Comment by jkosh44** [comment]
> That is, the failure is unrelated to your changes; you are good to proceed with the merge!

Thanks! I do need an explicit approval before merging if you're willing to give one.

**Review by nrainer-materialize — APPROVED** [review]
LGTM

**Pull request merged**
